File: frontend/src/utils/settlementTransactionUtils.ts
export const SETTLEMENT_TRANSACTION_TYPES = [
  'SHOP_TO_SYSTEM',
  'SYSTEM_TO_SHOP'] as const;
export const translateSettlementTransactionType = (value: string): string => {
  switch (value) {
    case 'SHOP_TO_SYSTEM': return 'Shop chuyển';
    case 'SYSTEM_TO_SHOP': return 'Hệ thống chuyển';
    default: return value;
  }
};


export const SETTLEMENT_TRANSACTION_STATUSES = [
  'PENDING',
  'SUCCESS',
  'FAILED'] as const;
export const translateSettlementTransactionStatus = (value: string): string => {
  switch (value) {
    case 'PENDING': return 'Đang xử lý';
    case 'SUCCESS': return 'Thành công';
    case 'FAILED': return 'Thất bại';
    default: return value;
  }
};

File: backend/src/main/java/com/logistics/utils/SettlementTransactionUtils.java
package com.logistics.utils;

import com.logistics.enums.SettlementTransactionStatus;
import com.logistics.enums.SettlementTransactionType;

public class SettlementTransactionUtils {

    public static String translateSettlementTransactionType(SettlementTransactionType value) {
        if (value == null) {
            return "";
        }

        return switch (value) {
            case SHOP_TO_SYSTEM -> "Shop chuyển";
            case SYSTEM_TO_SHOP -> "Hệ thống chuyển";
            default -> value.name();
        };
    }

    public static String translateSettlementTransactionStatus(SettlementTransactionStatus value) {
        if (value == null) {
            return "";
        }

        return switch (value) {
            case PENDING -> "Đang xử lý";
            case SUCCESS -> "Thành công";
            case FAILED -> "Thất bại";
            default -> value.name();
        };
    }
}